code/game/objects/items/weapons/implants/implant.dm

```diff
@@ -346,6 +346,26 @@ the implant may become unstable and either pre-maturely inject the subject or si
 			H << "<span class='notice'>You feel a surge of loyalty towards [company_name].</span>"
 		return 1
 
+	emp_act(severity)
+		if (malfunction)
+			return
+		malfunction = MALFUNCTION_TEMPORARY
+
+		activate("emp")
+		if(severity == 1)
+			if(prob(50))
+				meltdown()
+			else if (prob(50))
+				malfunction = MALFUNCTION_PERMANENT
+			processing_objects.Remove(src)

```

Comment:
Shouldn't the proc return here? Otherwise it'll revert malfunction to MALFUNCTION_TEMPORARY.